### PR TITLE
Javascript Classes & Inheritance

### PR DESCRIPTION
**Pull request by Skvesh**
1. Task:
   [Link to the current is here](https://github.com/rolling-scopes-school/tasks/compare/master...Skvesh:master)
2. App screenshot:
   image
4. Done 4.05.22 (deadline 9.05.22)
5. Details
    - [x] Base class ES5 is implemented
    - [x] Base class ES6is implemented
    - [x] Base subclass ES5 is implemented
    - [x] Base subclass ES6 is implemented

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+// "use strict";
+
+// ES5
+function Calculator(value) {
+	if (typeof(value) == 'number') {
```
Comment (gpukys) [inline script.js:5]: try to get into the habit of using `===`. There are very rare cases where you actually need to use the `==`

File: script.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+// "use strict";
+
+// ES5
+function Calculator(value) {
+	if (typeof(value) == 'number') {
+		this.int = value;
+	} else {
+		this.str = value
```
Comment (gpukys) [inline script.js:8]: Do we really need separate variables for storing the string or number? Why not just change this to `value` or something

File: script.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	}
+	return this;
+}
+StringBuilder.prototype.divide = function(n) {
+	const k = Math.floor(this.str.length / n);
+	this.str = this.str.slice(0, k);
+	return this;
+}
+StringBuilder.prototype.remove = function(str) {
+	this.str = this.str.split('').filter(char => char !== str).join('');
```
Comment (gpukys) [inline script.js:91]: what if `str` was a word, not a single character?

**Review by gpukys — CHANGES_REQUESTED** [review]

**Review by gpukys — APPROVED** [review]
Looks good to me

File: script.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+// ES5
+function Calculator(value) {
+	this.value = value;
+}
+Calculator.prototype.get = function() {
+	return this.value;
+}
+
+// ES6
+class newCalculator {
```
Comment (AlreadyBored) [inline script.js:12]: Classes names should start from capital letter

**Comment by AlreadyBored** [comment]
What is this at all? Are you sure that it should be here? :)

**Pull request closed without merging**